pkg/test-fw/ig/teststeps.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ig
+
+import (
+	"testing"
+	"time"
+)
+
+const stepWaitDuration = 10 * time.Second
+
+// TestStep allows combining different steps (e.g command, container creation)
+// to allow simplified/consistent flow for tests via RunTestSteps
+type TestStep interface {
+	// Run runs the step and wait its completion.
+	Run(t *testing.T)
+
+	// Start starts the step and immediately returns, it does wait until
+	// its completion, use Stop() for that.
+	Start(t *testing.T)
+
+	// Stop stops the step and waits its completion.
+	Stop(t *testing.T)
+
+	// IsCleanup returns true if the step is used to clean resource and
+	// should not be skipped even if previous commands failed.
+	IsCleanup() bool
```

Comment:
I think when we implemented these tests, we weren't aware of [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup), I think the can use it instead of this `IsCleanup()` logic.

(In any case, I think it's not used in this PR, so I think we can remove it).